### PR TITLE
ocamlPackages.bistro: unstable-2021-07-13 -> unstable-2021-11-13

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Motivation for this change

Unbreak in ocamlPackages_latest by merging upstream fixes.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [NA] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [NA] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [NA] (Module updates) Added a release notes entry if the change is significant
  - [NA] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/bistro/default.nix [hunk 1]
```diff
@@ -15,25 +16,28 @@
...
-  '';
-
   propagatedBuildInputs = [
-    base64 bos core lwt_react ocamlgraph ppx_sexp_conv rresult sexplib tyxml
+    base64
+    bos
+    core
+    lwt_react
+    ocamlgraph
+    ppxlib
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/bistro/default.nix:36]: Why this change? I think `ppxlib` should not be propagated.

Comment (bcdarwin) [reply pkgs/development/ocaml-modules/bistro/default.nix]: Reverted - upstream added it to their dependencies but I agree it probably doesn't make sense to propagate.
